### PR TITLE
Fixed docstring warnings and improvements

### PR DESCRIPTION
**Pull request by hassaanfarooq01**
- Fixed mkdocs docstring warnings
- Added improvements docstring texts

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@hassaanfarooq01 they're all fixed!!

**Comment by glenn-jocher** [comment]
@hassaanfarooq01 oh wait, this branch doesn't have the Docs checking action, so it's hard to tell haha. I think you're good to merge this into the CI branch, or vice versa, so you can tell if there are any warnings remaining.

**Comment by glenn-jocher** [comment]
@hassaanfarooq01 ok I just built locally. All the warnings are gone (good job)!

But there's one problem remaining. Every inlined return that does not have a variable name needs to be surrounded by parentheses in the docstrings, i.e.: here we need `(bool)` instead of just `bool`. Basically, types always need parentheses, otherwise `mkdocstrings` will not display them correctly (see screenshot) below:


```python
def example_function(arg1: int, arg2: int) -> bool:
    """
    Example function that demonstrates Google-style docstrings.

    Args:
        arg1 (int): The first argument.
        arg2 (int): The second argument.

    Returns:
        (bool): True if successful, False otherwise.

    Examples:
        >>> result = example_function(1, 2)  # returns False
    """
    if arg1 == arg2:
        return True
    return False
```

### Example screenshot showing problem
Screenshot 2024-01-24 at 18 49 24

**Comment by glenn-jocher** [comment]
Yeah for this it doesn't raise warnings unfortunately, it's hard to check automatically so often times I just spot the tables built incorrectly in the published docs and then go fix them in the `ultralytics` docs. 

Ok let me know when they're updated and I'll review again!

File: hub_sdk/modules/models.py [hunk 1]
```diff
@@ -170,79 +169,73 @@ def is_custom(self) -> bool:
...
+    def get_weights_url(self, weight: str = "best") -> Optional[str]:
         """
         Get the URL of the model weights.
 
         Args:
             weight (str, optional): Type of weights to retrieve. Defaults to "best".
 
         Returns:
-            str or None: The URL of the specified weights or None if not available.
+            Optional[str]: The URL of the specified weights or None if not available.
```
Comment (glenn-jocher) [inline hub_sdk/modules/models.py:202]: Needs parentheses

**Comment by hassaanfarooq01** [comment]
> @hassaanfarooq01 there are number of corrections required here, this is very far from ready to merge.
> 
> Most importantly ALL usage of Optional requires brackets rather than parentheses.
> 
> PyCharm displays this clearly in your PR, you should probably use it when viewing your code.
> 
> Screenshot 2024-01-25 at 13 57 26

ohhh! it stupid mistake. yeah using vscode will jump to pycharm.

**Comment by glenn-jocher** [comment]
@hassaanfarooq01 ok let me know when it's ready, thanks!

**Comment by hassaanfarooq01** [comment]
@glenn-jocher I have reviewed the documentation thoroughly and implement the necessary changes to address issues with ui, with that also tried to add Notes and proper descriptions that will be helpful visually.

**Comment by glenn-jocher** [comment]
@hassaanfarooq01 ok thanks, I will take a look today!

**Comment by glenn-jocher** [comment]
@hassaanfarooq01 looks much better! I think all the issues are resolved now. Nice work!

**Review by glenn-jocher — APPROVED** [review]

**Pull request merged**